app/pages/planner/page.tsx

```diff
@@ -80,13 +81,100 @@ export default function Planner() {
   const [isBrowser, setIsBrowser] = useState(false);
   const [numSelectedCourses, setNumSelectedCourses] = useState(0);
 
+  const [hasSaved, setHasSaved] = useState<boolean>(false);
+  const [called, setCalled] = useState<boolean>(true);
+  const [show, setShow] = useState<boolean>(false);
+
+  const animationProps = useSpring({
+    opacity: show ? 1 : 0,
+    transform: show ? 'transformX(0)' : 'transformX(-100%)',
+    onRest: () => setShow(false),
+    marginTop: '3px',
+  });
+
   // Required for SSR bug in react-beautiful-dnd
   useEffect(() => {
     if (typeof window !== 'undefined') {
       setIsBrowser(true);
     }
   }, []);
 
+  useEffect(() => {
+    if (called) {
+      handleGetCourses();
+      setCalled(false);
+    }
+  }, [called]);
+
+  //this pulls from backend and sets the hasSaved state to true if there is a saved course
+  const handleGetCourses = async () => {
+    try {
+      const response = await fetch('http://localhost:3000/api/saveCourses/GET', {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+
+      if (response.ok) {
+        const courses = await response.json();
+        console.log(courses);
+        setSelected({ course: courses.courses[0].saved_courses });
+        setNumSelectedCourses(courses.courses[0].saved_courses.length);
+        setHasSaved(true);
+      }
+    } catch (error) {
+      console.error('Error while getting the rating:', error);
+    }
+  };
+
+  const handleCourseSaving = async () => {
+    try {
+      // Create a data object with the required information
+      const data = {
+        courses: selected.course,
+      };
+
+      if (hasSaved) {
+        //update the saved courses instead of posting new one
+        const response = await fetch('http://localhost:3000/api/saveCourses/PUT', {
```

Comment:
Should we parameterise `'http://localhost:3000/api'` as a (global?) string somewhere? Then call

```ts
fetch(`${apiPath}/saveCourses/PUT`}
```

so that in theory if it were ever deployed you’d just have to change one variable as opposed to every `fetch()` call.

Reply:
While I’m here, what’s the rationale for putting the HTTP request method in the endpoint URI? (Versus just making `PUT` request to `/api/saveCourses`)

Reply:
The parameterising should be saved for a separate refactor branch if we do do it.  There is a lot of existing code that we inherited that referenced local host.

Reply:
It's the next.js structure the previous them decided to go with, a route for each type of request.

Reply:
Makes sense—thanks for filling me in!